src/main.c

```diff
@@ -29,15 +29,41 @@ int main(int argc, char *argv[])
 {
   LOG("STARTING SAT SOLVER", 3);
 
-  if (!check_args(argc, argv)) { return 0; }
-  
-  // TODO: malloc structs
+  FILE* fp = check_args(argc, argv);
+
+  if(!fp) 
+  { 
+  	LOG("Error while opening/reading file.", 3); 
+  	printf("ERROR\n");
+  	input_free(fp, NULL, NULL);
+  	return 0; 
+  }
   
-  if (!input_parser (NULL, NULL, NULL)) { printf("ERROR\n"); return 0; } // TODO fix args
-  if (!solve        (NULL, NULL      )) { printf("ERROR\n"); return 0; } // TODO fix args
-  if (!input_free   (NULL, NULL, NULL)) { printf("ERROR\n"); return 0; } // TODO fix args
+  UNMOLESTED_INPUT *unin = malloc(sizeof(UNMOLESTED_INPUT));
+  MOLESTED_INPUT *in = malloc(sizeof(MOLESTED_INPUT));
```

Comment:
Right now I'm only freeing 3 items (FILE* fp, unin, and in) but I'm going to add freeing for all the items that I've malloced in input.c so it'll get fairly lengthy and it would be nice to put that in a corner.